GooglePlayInstant/Editor/QuickDeploy/WwwRequestInProgress.cs

```
@@ -0,0 +1,110 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Runtime.CompilerServices;
+using UnityEditor;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("GooglePlayInstant.Tests.Editor.QuickDeploy")]
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// Provides functionality for tracking HTTP requests represented by corresponding WWW instances, as well as
+    /// executing scheduled delegages on the responses when the requests are complete.
+    /// </summary>
+    public class WwwRequestInProgress
+    {
+        private readonly WWW _www;
+        private readonly string _progressBarTitleText;
+
+
+        /// <summary>
+        /// A method to be executed on the _www field when it is done.
+        /// </summary>
+        public delegate void DoneWwwHandler(WWW www);
+
+        private DoneWwwHandler _onDone;
+
+        // Only one request can be performed at a time
+        private static WwwRequestInProgress _requestInProgress;
+
+
+        /// <summary>
+        /// Monitor an HTTP request is being made. Assumed one request will be made at a time, therefore will provide
+        /// a warning message and stop monitoring a previous request if one was still being monitored.
+        /// </summary>
+        /// <param name="www">A www instance holding a request that was made</param>
+        /// <param name="progressBarTitleText">A descriptive text to display as the title of the progress bar.</param>
+        /// <param name="onDone">A delegate to be invoked on the result once it is available.</param>
+        public static void TrackProgress(WWW www, string progressBarTitleText, DoneWwwHandler onDone)
+        {
+            if (_requestInProgress != null)
+            {
+                Debug.LogWarning("Started another request while the previous one was not complete.");
```

this should never happen right? Let's just throw an exception rather than hiding it

---

Resolved.